File: javascript/express/src/client/sunrisesunset.client.js
import axios from "axios";

export const getSunriseSunsetTimes = async (latitude, longitude) => {
  try {
    const sunriseAndSunset = await axios.get(
      `https://api.sunrise-sunset.org/json?tzid=Asia/Kolkata`,
      {
        params: {
          lat: latitude,
          lng: longitude,
        },
      }
    );
    return sunriseAndSunset.data;
  } catch (err) {
    console.error(err);
  }
};

File: python/sample-fastapi-api/app/clients/sunrisesunset_client.py
import httpx

URL = 'https://api.sunrise-sunset.org/json?tzid=Asia/Kolkata'


async def get_sunrise_sunset_async(latitude: str, longitude: str):
    params = {'lat': latitude, 'lng': longitude}
    async with httpx.AsyncClient() as client:
        response = await client.get(URL, params=params)
        return response.json() if response.status_code == 200 else {}
